docs/guides/serverless-hamsim-template.ipynb

```diff
@@ -0,0 +1,1087 @@
+    "\n",
+    "import numpy as np\n",
+    "import json\n",
+    "from mergedeep import merge\n",
+    "\n",
+    "\n",
+    "# Configure `EstimatorOptions`, to control the parameters of the hardware experiment\n",
+    "#\n",
+    "# Set default options\n",
+    "estimator_default_options = {\n",
+    "    \"resilience\": {\n",
+    "        \"measure_mitigation\": True,\n",
+    "        \"zne_mitigation\": True,\n",
+    "        \"zne\": {\n",
+    "            \"amplifier\": \"gate_folding\",\n",
+    "            \"noise_factors\": [1, 2, 3],\n",
+    "            \"extrapolated_noise_factors\": list(np.linspace(0, 3, 31)),\n",
+    "            \"extrapolator\": [\"exponential\", \"linear\", \"fallback\"],\n",
+    "        },\n",
+    "        \"measure_noise_learning\": {\n",
+    "            \"num_randomizations\": 512,\n",
+    "            \"shots_per_randomization\": 512,\n",
+    "        },\n",
+    "    },\n",
+    "    \"twirling\": {\n",
+    "        \"enable_gates\": True,\n",
+    "        \"enable_measure\": True,\n",
+    "        \"num_randomizations\": 300,\n",
+    "        \"shots_per_randomization\": 100,\n",
+    "        \"strategy\": \"active\",\n",
+    "    },\n",
+    "}\n",
+    "# Merge with user-provided options\n",
+    "estimator_options = merge(\n",
+    "    arguments.get(\"estimator_options\", {}), estimator_default_options\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "939b9b9c-34b7-40a1-92fc-2b1639aee349",
+   "metadata": {},
+   "source": [
+    "When the function template is running, it is helpful to return information in the logs by using print statements, so that you can better evaluate the workload's progress. Following is a simple example of printing the `estimator_options` so  there is a record of the actual Estimator options used. There are many more similar examples throughout the program to report progress during execution, including the value of the objective function during the iterative component of AQC-Tensor, and the two-qubit depth of the final instruction set architecture (ISA) circuit intended for execution on hardware."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d8d67489-9c92-42c0-a7fc-a347077ac72e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile --append ./source_files/function_template_hamsim.py\n",
+    "\n",
+    "print(\"estimator_options =\", json.dumps(estimator_options, indent=4))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a6b70531-d5c4-42c5-8fcc-9e19c32f32e8",
+   "metadata": {},
+   "source": [
+    "### Validate the inputs\n",
+    "\n",
+    "An important aspect of ensuring that the template can be reused across a range of inputs is to validate them. The following code is an example of verifying that the stopping fidelity during AQC-Tensor has been specified appropriately and if not, return an informative error message for how to fix the error."
```

Comment:
```suggestion
    "An important aspect of ensuring that the template can be reused across a range of inputs is input validation. The following code is an example of verifying that the stopping fidelity during AQC-Tensor has been specified appropriately and if not, returning an informative error message for how to fix the error."
```

Reply:
(I believe the "them" I removed was referring to the inputs - pls change if not)